### PR TITLE
fix(login): use label policy settings for favicon, translate titles

### LINKED ISSUE #4391: SEO texts for login and register
Users may want to customize their own SEO attributes for the hosted pages (excl. Console).
It's also unclear which language will be presented to robots.

image

**Acceptance criteria**
- [ ] Make Title meta-tags configurable
- [ ] Make Description meta-tags configurable

### LINKED ISSUE #4607: Branding: Adjust Favicon and page titles
The branding page misses the adaption of the favicon and the page titles are not translated.

**Acceptance criteria**
- The Logo defined in the branding page (Instanz Einstellungen --> Branding) is used as favicon
- The page title (<title>) appears in the right language

Bildschirmfoto 2022-10-25 um 11 39 25

### PR DESCRIPTION
**Pull request by peintnermax**
Closes #4607
Closes #4391

### CODE REVIEW

File: internal/api/ui/login/mfa_init_verify_handler.go [hunk 1]
```diff
@@ -70,7 +70,8 @@ func (l *Login) renderMFAInitVerify(w http.ResponseWriter, r *http.Request, auth
 	if err != nil {
 		errID, errMessage = l.getErrorMessage(r, err)
 	}
-	data.baseData = l.getBaseData(r, authReq, "MFA Init Verify", errID, errMessage)
+	translator := l.getTranslator(r.Context(), authReq)
+	data.baseData = l.getBaseData(r, authReq, "InitMFAPrompt.Title", "InitMFAPrompt.Description", errID, errMessage)
```
Comment (livio-a) [inline internal/api/ui/login/mfa_init_verify_handler.go:74]:
```suggestion
	data.baseData = l.getBaseData(r, authReq, "InitMFAOTP.Title", "InitMFAOTP.Description", errID, errMessage)
```

File: internal/api/ui/login/renderer.go [hunk 2]
```diff
@@ -330,15 +340,28 @@ func (l *Login) getUserData(r *http.Request, authReq *domain.AuthRequest, title
...
+	if (titleI18nKey != "") {
+		title = translator.LocalizeWithoutArgs(titleI18nKey)
+	}
+
+	description := ""
+	if (titleI18nKey != "") {
+		description =  translator.LocalizeWithoutArgs(descriptionI18nKey)
+	}
+	
+	lang, _ := l.renderer.ReqLang(translator, r).Base()
```
Comment (livio-a) [inline internal/api/ui/login/renderer.go:356]: i'd just pass the i18n key to the html and let i translate there

File: internal/api/ui/login/renderer.go [hunk 3]
```diff
@@ -330,15 +340,28 @@ func (l *Login) getUserData(r *http.Request, authReq *domain.AuthRequest, title
...
 	baseData := baseData{
 		errorData: errorData{
 			ErrID:      errType,
 			ErrMessage: errMessage,
 		},
 		Lang:                   lang.String(),
 		Title:                  title,
+		Description:			description,
```
Comment (livio-a) [inline internal/api/ui/login/renderer.go:364]:
```suggestion
		Description:		description,
```

File: internal/api/ui/login/static/i18n/en.yaml [hunk 4]
```diff
@@ -284,7 +284,7 @@ LinkingUsersDone:
   NextButtonText: next
 
 ExternalNotFound:
-  Title: External User
+  Title: External User not found
```
Comment (livio-a) [inline internal/api/ui/login/static/i18n/en.yaml:287]: what about the other languages?

File: internal/api/ui/login/static/templates/main.html [hunk 5]
```diff
@@ -8,13 +8,24 @@
     <meta http-equiv="X-UA-Compatible" content="ie=edge">
 
     <link rel="stylesheet" href="{{ resourceThemeUrl "css/zitadel.css" .Theme }}" type="text/css">
-    <link rel="icon" type="image/x-icon" href="{{ resourceThemeUrl "favicon.ico" .Theme }}">
+    
     {{ if hasCustomPolicy .LabelPolicy }}
     <link rel="stylesheet" href="{{ variablesCssFileUrl .PrivateLabelingOrgID .LabelPolicy}}" type="text/css">
     {{ end}}
+
+    {{ if hasCustomPolicy .LabelPolicy }}
```
Comment (livio-a) [inline internal/api/ui/login/static/templates/main.html:16]: there's the exact same check just above... i'd merge them

File: internal/i18n/i18n.go [hunk 6]
```diff
@@ -139,6 +139,10 @@ func (t *Translator) Localize(id string, args map[string]interface{}, langs ...s
 	return localize(t.localizer(langs...), id, args)
 }
 
+func (t *Translator) LocalizeWithoutArgs(id string, langs ...string) string {
+	return localize(t.localizer(langs...), id, map[string]interface{}{})
+}
+
```
Comment (livio-a) [inline internal/i18n/i18n.go:145]: not necessary if i18n keys are passed to the html

**Review by livio-a — CHANGES_REQUESTED** [review]

File: internal/api/ui/login/renderer.go (hunk 2 above)
Comment (livio-a) [reply internal/api/ui/login/renderer.go:356]: hmm on a second thought, checking for empty strings saves us possible warnings

but then you will need to check for the correct key `if descriptionI18nKey != ""` and please omit the brackets

File: internal/api/ui/login/renderer.go [hunk 7]
```diff
@@ -315,13 +325,13 @@ func (l *Login) renderInternalError(w http.ResponseWriter, r *http.Request, auth
 	if err != nil {
 		_, msg = l.getErrorMessage(r, err)
 	}
-	data := l.getBaseData(r, authReq, "Error", "Internal", msg)
+	data := l.getBaseData(r, authReq, "Error","", "Internal", msg)
```
Comment (livio-a) [inline internal/api/ui/login/renderer.go:328]: if we change to i18n keys we either need one for `Error` or we should change to `Errors.Internal`

File: internal/api/ui/login/static/templates/main.html (hunk 5 above)
Comment (peintnermax) [reply internal/api/ui/login/static/templates/main.html]: done

File: internal/api/ui/login/renderer.go (hunk 7 above)
Comment (peintnermax) [reply internal/api/ui/login/renderer.go]: changed to `Errors.Internal`

File: internal/api/ui/login/static/i18n/en.yaml (hunk 4 above)
Comment (peintnermax) [reply internal/api/ui/login/static/i18n/en.yaml:287]: done

File: internal/i18n/i18n.go (hunk 6 above)
Comment (peintnermax) [reply internal/i18n/i18n.go:145]: kept it because its still checked for empty in the renderer..

File: internal/api/ui/login/renderer.go (hunk 2 above)
Comment (peintnermax) [reply internal/api/ui/login/renderer.go]: changed checks 👍

**Review by livio-a — APPROVED** [review]

**Pull request merged**